### ISSUE TITLE
Unexpected panic during doc tests, Failed to get crate data

### ISSUE THREAD

**Issue by ralpha** [rust-lang/rust#85127]
### Code

It crashed on a docs tests, but there is no doc-test in that crate.
So there is no real way of adding code.

### Meta
I was not able to reproduce the error.
Running the command again (`$ cargo test`) did not give any errors.
(although it did recompile some crates because I was still changing thing, mainly normal comments, not doc-comments)

`rustc --version --verbose`:
```
rustc 1.52.0 (88f19c6da 2021-05-03)
binary: rustc
commit-hash: 88f19c6dab716c6281af7602e30f413e809c5974
commit-date: 2021-05-03
host: x86_64-unknown-linux-gnu
release: 1.52.0
LLVM version: 12.0.0
```

### Error output

```
   Doc-tests df_ls_semantic_analysis
thread 'rustc' panicked at 'Failed to get crate data for crate18', compiler/rustc_metadata/src/creader.rs:136:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.52.0 (88f19c6da 2021-05-03) running on x86_64-unknown-linux-gnu

note: compiler flags: -C embed-bitcode=no --crate-type lib

query stack during panic:
end of query stack
error: test failed, to rerun pass '--doc'
```

Output after rerun of command (was also expecting this output):
```
   Doc-tests df_ls_syntax_analysis

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

### Other info
I don't know if this is relevant to this, but in case it is.
When I run tests it all works very fast, unit/integration tests all finish in about 3-4 sec. (about 140 tests)
But the last doc tests (about 4, VERY small, tests) take about 6 sec to finish, even on crates that don't have any docs-tests in them I can still see it slow down a lot.

Might be related to: #84049, #84026

**Comment by Enselic** [comment]
Triage: Without a way for others to reproduce the ICE, not much can be done at this point. Closing.
